src/api/ZoweExplorerZosmfApi.ts

```diff
@@ -36,23 +37,39 @@ class ZosmfApiCommon implements ZoweExplorerApi.ICommon {
         return ZosmfUssApi.getProfileTypeName();
     }
 
-    public getSession(profile?: IProfileLoaded): Session {
+    public async getSession(profile?: IProfileLoaded): Promise<Session> {
         if (!this.session) {
-            this.session = zowe.ZosmfSession.createBasicZosmfSession((profile||this.profile).profile);
+            this.session = await getValidSession((profile||this.profile), (profile||this.profile).name, false);
```

Comment:
Since getSession() calls getValidSession() why not have calls that deal with all profile types throughout ZE just call getSession() and for type zosmf it will call the getValidSession() method and handle it while other extension will continue to handle theirs with their api call of getSession()?

Reply:
> Since getSession() calls getValidSession() why not have calls that deal with all profile types throughout ZE just call getSession() that is in each type's api file and for type zosmf it will call the getValidSession() method and handle it while other extension will continue to handle theirs with their api call of getSession()?

@katelynienaber I think this may be the best way to go. It would give you the flexibility in this PR to handle `getValidSession()` the way you need to for zosmf profiles, while also allowing extenders to handle `getSession()` the way they need to.

Reply:
I am not sure this would work. I need to be able to control if prompting occurs, which I can pass into getValidSession, but not getSession.

I guess I can add it as an optional parameter? Would this still break extenders?

Also importing `ZoweExplorerApiRegister` everywhere might cause this circular dependency again >.< We can try though. And I believe I had some trouble with `ZoweExplorerApiRegister.getMvsApi(profile)` in some places because the profile was not available for some reason...but again we can try and see what comes up.

Reply:
> I guess I can add it as an optional parameter? Would this still break extenders?

New optional parameters should be fine 😋